libvast/src/chunk.cpp

```diff
@@ -147,6 +148,20 @@ chunk_ptr chunk::slice(view_type view) const {
   });
 }
 
+// -- free functions ----------------------------------------------------------
+
+std::shared_ptr<arrow::Buffer> as_arrow_buffer(chunk_ptr chunk) noexcept {
+  if (!chunk)
+    return nullptr;
+  auto buffer = arrow::Buffer::Wrap(chunk->data(), chunk->size());
```

Comment:
I'm a bit confused by the arrow docs, which state that

> The Buffer base class does not own its memory, but subclasses often do.

On the other hand, the API seems to be written in a way that implies that a `Buffer` may own its memory. Since arrow itself provides classes like `PyBuffer` and `NumPyBuffer`, I wonder if the intended usage would be to create a `VastChunkBuffer` subclass?

Reply:
`arrow::Buffer` is essentially what `vast::chunk` is: A view on a contiguous byte range with a deleter. The `std::shared_ptr` / `caf::intrusive_ptr` wrapping only serves refcounting purposes, and both classes are designed to be never be copied.

For `chunk`, we have the additional restrictions that (1) the memory is read-only, that (2) a sliced chunk does not remember its parents, and that (3) a chunk does type-erasure and we cannot retrieve the typed contiguous byte range it was created from.

The `Buffer` itself never owns memory, but you often work with sub classes of `Buffer` that do, and are then accessed via the base classes interface. One way to implement this function would be to, say, create this `ChunkBuffer` subclass that actually shares the refcount of the `Buffer`. This would serve the same purpose, but then we'd need to implement a bunch more functions from the `Buffer` interface.

Reply:
Sounds reasonable. From looking at `arrow/buffer.hpp`  doesn't seem to be doing virtual functions but rather let subclasses control its behavior by setting the protected member functions.

I don't really have a strong opinion on whether a subclass is a good idea or not.